src/clients.rs

```diff
@@ -202,24 +203,19 @@ struct InnerState {
 
 impl State {
     /// Creates a new state instance. Should only be called once.
-    ///
-    /// # Panics
-    /// If called before [`set_cache_dir`].
-    pub fn new(client: RuskStateClient) -> Result<Self, Error> {
-        let cache = Cache::new()?;
+    pub fn new(
+        client: RuskStateClient,
+        data_dir: &Path,
+    ) -> Result<Self, Error> {
+        let cache = Cache::new(data_dir)?;
         let inner = Mutex::new(InnerState { client, cache });
```

Comment:
As the cache instance is wrapped in a Mutex here,  do we expect any multi-threaded calls to `cache.insert`?

Asking this, cause we may end up having a race condition over `last_height` mutation.

Reply:
@goshawk-3 AFAIK The cache inserts are only called from `fetch_notes` which may be called by this async function here https://github.com/dusk-network/wallet-cli/blob/main/src/bin/command.rs#L338 

@herr-seppia or @ZER0 might wanna elaborate on why is `inner` state wrapped in a `Mutex`?

The client was wrapped in a `Mutex` so I also put the cache in there. It might not supposed to be in a `Mutex`.

Reply:
Just for the record, could you confirm the scenario below cannot happen in any case?

**Persisted state:
Cf: my_key
last_height = 1**

Scenario:

1. write_batch_1 = insert( my_key, last_height=4)
2. write_batch_2 = insert( my_key, last_height=3)
3. write_batch_1.persist()
4. write_batch_2.persist()

Persisted state:
Cf: my_key
last_height = 3 **(instead of 4)**

Reply:
I think we're good with race conditions, the Put(s) and Get(s) in the rocksdb wiki are labelled as "thread-safe" unless we are using unsafe rust, (which we are not and rocksdb crate is a well used and maintained crate)  I think we are good
https://github.com/facebook/rocksdb/wiki/RocksDB-FAQ#basic-readwrite

Reply:
As explained already, it's not about thread-safety property of RocksDB but the order of execution of data persistence.  

If there is only one caller of insert-persist and it's blocking then the order is guaranteed.